src/networks.jl

```diff
@@ -994,3 +994,262 @@ Carries on the spatial convolution respecting periodicity at the boundaries.
+              denset2=Dense(embed_dim, channels[1]),
+              control_denset2=Dense(embed_dim, channels[1]),
+              tgnorm2=GroupNorm(channels[1], 32, swish),
+              
+              # Projection
+              tconv1=tconv1,
+              mean_bypass_layers...
+              )
+
+    return ControlledNoiseConditionalScoreNetwork(layers, control_net, mean_bypass, scale_mean_bypass, shift_input, shift_output, gnorm)
+end
+
+@functor ControlledNoiseConditionalScoreNetwork
+
+"""
+    (net::ControlledNoiseConditionalScoreNetwork)(x, c, t)
+
+Evaluates the neural network of the NoiseConditionalScoreNetwork
+model on (x,c,t), where `x` is the tensor of noised input,
+`c` is the tensor of contextual input, and `t` is a tensor of times.
+"""
+function (net::ControlledNoiseConditionalScoreNetwork)(x, c, t)
+    # Get size of spatial dimensions
+    nspatial = ndims(x) - 2
+
+    # Embeddings
+    embed = net.layers.gaussfourierproj(t)
+    embed = net.layers.linear(embed)
+    control_embed = net.control_net(c)
+
+    # Encoder
+    if net.shift_input
+        h1 = x .- mean(x, dims=(1:nspatial)) # remove mean of noised variables before input
+    else
+        h1 = x
+    end
+    h1 = net.layers.conv1(h1)
+    h1 = h1 .+ expand_dims(net.layers.dense1(embed) .+ net.layers.control_dense1(control_embed), nspatial)
+    h1 = net.layers.gnorm1(h1)
+    h2 = net.layers.conv2(h1)
+    h2 = h2 .+ expand_dims(net.layers.dense2(embed) .+ net.layers.control_dense2(control_embed), nspatial)
+    h2 = net.layers.gnorm2(h2)
+    h3 = net.layers.conv3(h2)
+    h3 = h3 .+ expand_dims(net.layers.dense3(embed) .+ net.layers.control_dense3(control_embed), nspatial)
+    h3 = net.layers.gnorm3(h3)
+    h4 = net.layers.conv4(h3)
+    h4 = h4 .+ expand_dims(net.layers.dense4(embed) .+ net.layers.control_dense4(control_embed), nspatial)
+
+    # middle
+    h = h4
+    for block in net.layers.resnet_blocks
+        h = block(h, embed .+ control_embed) # add in control embedding, can perhaps be done better.
+    end
+
+    # Decoder
+    h = net.layers.gnorm4(h)
+    h = net.layers.tconv4(h)
+    h = h .+ expand_dims(net.layers.denset4(embed) .+ net.layers.control_denset4(control_embed), nspatial)
+    h = net.layers.tgnorm4(h)
+    h = net.layers.tconv3(cat(h, h3; dims=nspatial+1))
+    h = h .+ expand_dims(net.layers.denset3(embed) .+ net.layers.control_denset3(control_embed), nspatial)
+    h = net.layers.tgnorm3(h)
+    h = net.layers.tconv2(cat(h, h2, dims=nspatial+1))
+    h = h .+ expand_dims(net.layers.denset2(embed) .+ net.layers.control_denset2(control_embed), nspatial)
+    h = net.layers.tgnorm2(h)
+    h = net.layers.tconv1(cat(h, h1, dims=nspatial+1))
+    if net.shift_output
+        h = h .- mean(h, dims=(1:nspatial)) # remove mean after output
+    end
+
+    # Mean processing of noised variable channels
+    if net.mean_bypass
```

Comment:
should the embedded control/condition also be used by the mean bypass layer?

Reply:
(we do use the context in the mean bypass in our other version, though it's only the mean of the context that is used)